src/qvcocoafunctions.mm

```diff
@@ -65,16 +65,33 @@ static void fixNativeMenuEccentricities(QMenu *menu, NSMenu *nativeMenu)
 }
 
 // This function should only be called once because it sets observers
-void QVCocoaFunctions::setFullSizeContentView(QWindow *window)
+void QVCocoaFunctions::setFullSizeContentView(QWindow *window, const bool shouldEnable)
 {
     auto *view = reinterpret_cast<NSView*>(window->winId());
 
-    // If this Qt and macOS version combination is already using layer-backed view, then enable full size content view
-    if (view.wantsLayer)
+    const bool isAlreadyEnabled = view.window.styleMask & NSWindowStyleMaskFullSizeContentView;
```

Comment:
Not sure what to put for comments here, I think the ones I just added in mainwindow covers it. Previously this function could only enable full size content view; the changes here are to allow it to be disabled as well. The isAlreadyEnabled stuff is just to protect against double calls (e.g. trying to disable it if it's already disabled).

Reply:
It would be good to be able to get a bit more info about what this function needs to do in its own context. It would be good to document:
- the `isAlreadyEnabled` you mentioned
- what is the purpose of `shouldEnable`? 
- What is the purpose of doing the `setFrame` situation?

Reply:
I cleaned things up a bit and added comments. I'm not even sure what the `titlebarOverlap` part was about anymore, I believe it was not strictly for the geometry save/restore workaround but somehow needed for the macOS session state feature I developed around the same time to work properly. Although even with that feature I'm not seeing problems by removing it, I probably eliminated the need for it with subsequent changes I made.